packages/main/src/DateComponentBase.ts

```diff
@@ -125,6 +133,24 @@ class DateComponentBase extends UI5Element {
 		return this._formatPattern !== "medium" && this._formatPattern !== "short" && this._formatPattern !== "long";
 	}
 
+	_getMinMaxCalendarDateFromString(date: string) {
+		// <b>Note:</b> Format#parse accepts only boolean type for 2nd and 3rd params,
+		// but has logic related to "undefined" value, so we're calling it with "undefined" and casting to "boolean".
+		const utc = undefined as unknown as boolean;
+		const strict = undefined as unknown as boolean;
+
+		if (this.getFormat().parse(date, utc, strict)) {
+			return this._getCalendarDateFromString(date);
+		}
+
+		const jsDate = this.getISOFormat().parse(date, utc, strict) as Date;
```

Comment:
In cases when we have pattern including intervals (e.g. 2023-03-31 - 2023-05-02). Which in our case does not concern us, because of the patternn we have provided and the strict parsing.